src/basic/virt.c

```diff
@@ -184,23 +194,44 @@ static int detect_vm_dmi(void) {
                         return r;
                 }
 
+
+
                 for (j = 0; j < ELEMENTSOF(dmi_vendor_table); j++)
-                        if (startswith(s, dmi_vendor_table[j].vendor))
+                        if (startswith(s, dmi_vendor_table[j].vendor)) {
+                                log_debug("Virtualization %s found in DMI (%s)", s, dmi_vendors[i]);
                                 return dmi_vendor_table[j].id;
+                        }
         }
 #endif
 
+        log_debug("No virtualization found in DMI");
+
         return VIRTUALIZATION_NONE;
 }
 
 static int detect_vm_xen(void) {
+        /* Check for Dom0 will be executed later in detect_vm_xen_dom0
+           Thats why we dont check the content of /proc/xen/capabilities here. */
+        if (access("/proc/xen/capabilities", F_OK) < 0) {
+                log_debug("Virtualization XEN not found, /proc/xen/capabilities does not exist");
+                return VIRTUALIZATION_NONE;
+        }
+
+        log_debug("virtualization XEN found (/proc/xen/capabilities exists)");
```

Comment:
Done.

 

Von: Zbigniew Jędrzejewski-Szmek [mailto:notifications@github.com] 
Gesendet: Dienstag, 23. Februar 2016 20:38
An: systemd/systemd <systemd@noreply.github.com>
Cc: nafets227 <infos@nafets.de>
Betreff: Re: [systemd] Bugfix set virt=none when running on XEN Dom0 (#2639)

 

In src/basic/virt.c <https://github.com/systemd/systemd/pull/2639#discussion_r53836613> :

>          return VIRTUALIZATION_NONE;
>  }
>  
>  static int detect_vm_xen(void) {
> +        /* Check for Dom0 will be executed later in detect_vm_xen_dom0
> +           Thats why we dont check the content of /proc/xen/capabilities here. */
> +        if (access("/proc/xen/capabilities", F_OK) < 0) {
> +                log_debug("Virtualization XEN not found, /proc/xen/capabilities does not exist");
> +                return VIRTUALIZATION_NONE;
> +        }
> +
> +        log_debug("virtualization XEN found (/proc/xen/capabilities exists)");

"V"

—
Reply to this email directly or view it on GitHub <https://github.com/systemd/systemd/pull/2639/files#r53836613> .  <https://github.com/notifications/beacon/AFW-psINTVUkngSobPuWoVyrfX3p2gRkks5pnKyogaJpZM4Hbhav.gif>